rasa_sdk/knowledge_base/actions.py

```diff
@@ -0,0 +1,290 @@
+        # entity mentioned in the conversation
+        return tracker.get_slot(SLOT_LAST_ENTITY)
+
+    def _resolve_mention(self, tracker):
+        """
+        Resolves a mention of an entity, such as first, to the actual entity.
+        If multiple entities are listed during the conversation, the entities
+        are stored in the slot 'knowledge_base_entities' as a list. We resolve the
+        mention, such as first, to the list index and retrieve the actual entity.
+        If the mention is not an ordinal mention, but some other reference, we just
+        assume the user is referring to the last mentioned entity in the conversation.
+
+        Args:
+            tracker: the tracker
+
+        Returns: name of the actually entity
+        """
+
+        mention = tracker.get_slot(SLOT_MENTION)
+        listed_items = tracker.get_slot(SLOT_LISTED_ITEMS)
+        last_entity = tracker.get_slot(SLOT_LAST_ENTITY)
+        last_entity_type = tracker.get_slot(SLOT_LAST_ENTITY_TYPE)
+        current_entity_type = tracker.get_slot(SLOT_ENTITY_TYPE)
+
+        if not mention:
+            return None
+
+        if listed_items and mention in self.knowledge_base.ordinal_mention_mapping:
+            idx_function = self.knowledge_base.ordinal_mention_mapping[mention]
+            return idx_function(listed_items)
+
+        # NOTE:
+        # for now we just assume, that if the user refers to an entity, for
+        # example via "it" or "that restaurant". he actually refers to the last
+        # entity that was detected.
+        if current_entity_type == last_entity_type:
+            return last_entity
+
+    def _to_str(self, entity_type, entity):
+        """
+        Converts an entity to its string representation using the lambda function
+        defined in the schema
+
+        Args:
+            entity_type: the entity type
+            entity: the entity with all its attributes
+
+        Returns: a string that represents the entity
+        """
+        representation_func = self.knowledge_base.schema[entity_type][
+            SCHEMA_KEYS_REPRESENTATION
+        ]
+        return representation_func(entity)
+
+    def _get_attributes_of_entity(self, entity_type, tracker):
+        """
+        Checks if the NER found any value for all attributes of the given entity type.
+
+        Args:
+            entity_type: the entity type
+            tracker: the tracker
+
+        Returns: a list of attributes
+        """
+        attributes = []
+
+        if entity_type not in self.knowledge_base.schema:
+            return attributes
+
+        for attr in self.knowledge_base.schema[entity_type][SCHEMA_KEYS_ATTRIBUTES]:
+            attr_val = tracker.get_slot(attr)
+            if attr_val is not None:
+                attributes.append({"name": attr, "value": attr_val})
+
+        return attributes
+
+    def _reset_attribute_slots(self, entity_type, tracker):
```

Comment:
it's not obvious to me why we need this

Reply:
If the user is saying something like "show me all restaurants with Italian cuisine", the NER should detect `restaurant` as `entity_type` and `Italian` as `cuisine` entity. So, we should filter the restaurant entities by their cuisine (= Italian). 
When listing entities, we check what attributes are detected by the NER. We take all attributes that are set, e.g. `cuisine=Italian`. If we don't reset the attribute slots after the request is done and the next utterance of the user would be, for example, "list all restaurants that have wifi", we now have two attribute slots set: `wifi` and `cuisine`. Thus, we would filter all restaurants for two attributes now: `wifi=True` and `cuisine=Italian`. To avoid that we reset the attribute slots once the request is done.

Reply:
ok! I suspected this was the case, but that should be documented in the docstring & docs